rackspace/autoscale/v1/launch/results.go

```diff
@@ -0,0 +1,353 @@
+}
+
+// ConfigurationType represents a type of launch configuration.
+type ConfigurationType string
+
+// Valid launch configuration types.
+const (
+	LaunchServer ConfigurationType = "launch_server"
+)
+
+// Configuration represents an auto scale group's launch configuration.
+type Configuration struct {
+	// Type for this configuration.
+	Type ConfigurationType `mapstructure:"type" json:"type"`
+
+	// Configuration arguments.
+	Args Args `mapstructure:"args" json:"args"`
+}
+
+// Validate ensures a Configuration contains appropriate values.
+func (c *Configuration) Validate() error {
+	if c.Type != LaunchServer {
+		return ErrUnknownType
+	}
+
+	if err := c.Args.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Args represents a launch configuration's arguments.
+type Args struct {
+	// Attributes used to create new servers in the group.
+	Server Server `mapstructure:"server" json:"server"`
+
+	// List of load balancers to which to attach servers.
+	LoadBalancers []LoadBalancer `mapstructure:"loadBalancers" json:"loadBalancers"`
+
+	// Number of seconds a deleted node is put into DRAINING mode in attached
+	// load balancers before actually being removed.  This will be zero if no
+	// timeout as been specified, in which case nodes will not be drained.
+	DrainingTimeout int `mapstructure:"draining_timeout" json:"draining_timeout,omitempty"`
+}
+
+// Validate ensures an Args struct contains appropriate values.
+func (a *Args) Validate() error {
+	if err := a.Server.Validate(); err != nil {
+		return err
+	}
+
+	for _, lb := range a.LoadBalancers {
+		if err := lb.Validate(); err != nil {
+			return err
+		}
+	}
+
+	if a.DrainingTimeout != 0 && (a.DrainingTimeout < 30 || a.DrainingTimeout > 3600) {
+		return ErrBadDrainTimeout
+	}
+
+	return nil
+}
+
+// mapstructure.DecodeHookFuncType to convert from a map[string]interface{} to a
+// servers.File, decoding contents as base64 if possible.
+func mapToFile(from reflect.Type, to reflect.Type, data interface{}) (interface{}, error) {
+	fileType := reflect.TypeOf((*servers.File)(nil))
+
+	if from.Kind() != reflect.Map || to != fileType {
+		return data, nil
+	}
+
+	fileMap, ok := data.(map[string]interface{})
+
+	if !ok {
+		return data, nil
+	}
+
+	file := servers.File{}
+
+	// If the path key exists and is a string, set file.Path.
+	if p, ok := fileMap["path"].(string); ok {
+		file.Path = p
+	}
+
+	// If the contents key exists and is a string, decode the string as base64
+	// and set file.Contents.  If decoding fails, use the raw bytes.
+	if c, ok := fileMap["contents"].(string); ok {
+		bytes, err := base64.StdEncoding.DecodeString(c)
+
+		if err == nil {
+			file.Contents = bytes
+		} else {
+			file.Contents = []byte(c)
+		}
+	}
+
+	return &file, nil
+}
+
+// mapstructure.DecodeHookFuncType to convert from a string to []byte decoding
+// contents as base64 if possible.
+func stringToBytes(from reflect.Type, to reflect.Type, data interface{}) (interface{}, error) {
```

Comment:
This works well for decoding `user_data`, but it's unfortunate that mapstructure then seems to run the hooks for each resulting byte.